blocks/form/form.js

```diff
@@ -0,0 +1,433 @@
+function formatNumberPromise(inputs, format) {
+  let arrayInputs = inputs;
+  if (!(inputs instanceof Array)) {
+    arrayInputs = [inputs];
+  }
+  if (!format) {
+    return Promise.resolve(arrayInputs);
+  }
+  function formatFn(formatters) {
+    return arrayInputs.map((num) => {
+      try {
+        return formatters.default(num, format);
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.log(`unable to format ${num} with format ${format}`);
+        return num;
+      }
+    });
+  }
+  function errorFn(e) {
+    // eslint-disable-next-line no-console
+    console.log('error in obtaining formatters', e);
+    return arrayInputs;
+  }
+  return import('./formatting.js').then(formatFn, errorFn);
```

Comment:
some comment would be useful here.... however, I don't think this is correct all the time. you relying on the fact that that `import` is always async, which it might not be, if the module is already loaded.

maybe something like:

```suggestion
  const formatNumber = await import('./formatting.js');
  return new Promise((resolve) => setImmediate(() => formatNumber());
```

Reply:
@tripodsan `import()` returns a [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import#return_value), so the `.then()` call should work, even if we are dealing with a promise that is resolved immediately.

Reply:
I discussed with customer. They do not need the full Intl support and have the following requirements

1. grouping symbol and decimal symbol will always be `,`  and `.` 
2. The currency symbol will change once they move to other south african countries and there will be only limited number of countries they think.
3. They do not need to format `years` and `months` in different language.

I already found a formatting function which performs almost equivalent of Intl (except the initial lag) [1] given the set of constraints the customer has

[1] https://jsfiddle.net/vdua/mpodxt4e/1/